bob-backend/src/interval_logger.rs

```diff
@@ -0,0 +1,60 @@
+use coarsetime::{Duration as CDuration, Instant as CInstant};
+use std::{
+    collections::HashMap,
+    hash::Hash,
+    cmp::Eq,
+    fmt::Display
+};
+use parking_lot::Mutex as PLMutex;
+
+#[derive(Debug)]
+pub struct IntervalErrorLogger<E> {
+    errors: HashMap<E, u64>,
+    interval: CDuration,
+    last_timestamp: CInstant,
+}
+
+impl<E: Hash + Eq + Display> IntervalErrorLogger<E> {
+    pub fn new(interval_ms: u64) -> Self {
+        Self {
+            errors: HashMap::new(),
+            interval: CDuration::from_millis(interval_ms),
+            last_timestamp: CInstant::now(),
```

Comment:
I suggest subtracting the `interval` to ensure that the very first message is always logged